include/jemalloc/internal/mutex_pool.h

```diff
@@ -60,27 +60,26 @@ mutex_pool_postfork_child(tsdn_t *tsdn, mutex_pool_t *pool) {
 
 static inline void
 mutex_pool_lock(tsdn_t *tsdn, mutex_pool_t *pool, uintptr_t key) {
-	mutex_pool_assert_not_held(tsdn, pool);
 
```

Comment:
What about instead having each thread keep a few spare pointers to mutex_pool_ts (and a lock count too maybe) in its witness data, and we can just assert that the pool we're locking isn't already there.

Locking pool mutexes incrementally is a fairly easy-to-write but subtle bug that will usually just never show up in testing. If we're expanding pool uses, it'd be nice to know we can't introduce those sorts of bugs.

Reply:
sure, sounds reasonable